CNKI.js

```diff
@@ -252,6 +262,26 @@ function scrape(ids, doc, url, itemInfo) {
 			//	newItem.extra = 'CN ' + newItem.callNumber;
 				newItem.callNumber = "";
 			}
+			// 018<lyb018@gmail.com>: 20210108
+			var num = doc.querySelector('#func3 .num');
+			if (!num) {
+				var e = doc.querySelector('[value="' + dbname + '!' + filename + '!' + curRec + '!' + queryID + '"]');
+				if (e) {
+					e = e.parentElement;
+				}
+				if (e) {
+					e = e.parentElement;
+				}
+				if (e) {
+					e = e.querySelector('.quote a');
+				}
+				if (e) {
+					num  = e
```

Comment:
This is a bit of a DOM roller-coaster ride! Is there a more reliable way to get to the element you want? It just feels likely to break quickly.

Reply:
I have fixed it and resubmitted it. Please review it again. Thank you.